app/controllers/roles_controller.rb

```diff
@@ -81,10 +80,17 @@ def update
 
   def destroy
     @role = Role.find(params[:id])
+    # after destroy permissions can not be reached
+    permissions = @role.permissions
     @role.destroy
+
     flash[:notice] = I18n.t(:notice_successful_delete)
     redirect_to action: 'index'
-    notify_changed_roles(:removed, @role)
+
+    OpenProject::Notifications.send(
+      OpenProject::Events::ROLE_DESTROYED,
+      permissions:
+    )
```

Comment:
Not mandatory but it would be an improvement if all of this would be handled by a DeleteService.